accepted/future-releases/class-modifiers/feature-specification.md

```diff
@@ -631,127 +624,82 @@ It is a compile-time error to:
     class C3 implements SM {} // Error.
     ```
 
-*   Mix in a mixin or mixin class marked `interface`, `final` or `sealed`
-    outside of the library where it is declared.
-
-    ```dart
-    // a.dart
-    interface mixin class I {}
-    final mixin class F {}
-    sealed mixin class S {}
-
-    interface mixin IM {}
-    final mixin FM {}
-    sealed mixin SM {}
+*   Use the interface of a class declaration marked `sealed`
+    as a mixin `on` type outside of library where it is declared.
 
-    // b.dart
-    import 'a.dart';
+_An `enum` declaration still cannot be implemented, extended or mixed in
+anywhere, independently of modifiers._
 
-    class C1 with I {} // Error.
-    class C2 with F {} // Error.
-    class C3 with S {} // Error.
+A type alias (`typedef`) cannot be used to subvert these restrictions
+or any of the restrictions below. When extending, implementing, or mixing in
+a type alias, we look at the library where class or mixin the type alias
+resolves to is defined to determine if the behavior is allowed. *Note that
+the library where the _type alias_ is defined does not come into play.
+Type aliases cannot be marked with any of the new modifiers.*
 
-    class C1 with IM {} // Error.
-    class C2 with FM {} // Error.
-    class C3 with SM {} // Error.
-    ```
+### Disallowing implementation
 
-A typedef cannot be used to subvert these restrictions or any of the
-restrictions below. When extending, implementing, or mixing in a typedef, we
-look at the library where class or mixin the typedef resolves to is defined to
-determine if the behavior is allowed. *Note that the library where the _typedef_
-is defined does not come into play. Typedefs cannot be marked with any of the
-new modifiers.*
+We say that an interface _prevents implementation_ if it has *any*
+superinterface whose declaration is marked `base` or `final`.
 
-### Disallowing implementation
+It is a compile-time error if a the interface of a
+`class`, `mixin`, or `mixin class` declaration `D` prevents implementation,
+and `D` does not have a `base`, `final` or `sealed` modifier.
+_(Which must be a `base` modifier for `mixin` and `mixin class`
+declarations.)_
 
-It is a compile-time error if a subtype of a declaration marked `base` or
-`final` is not marked `base`, `final`, or `sealed`. This restriction applies to
-both direct and indirect subtypes and along all paths that introduce subtypes:
+_Effectively, it is a compile-time error if any subtype of a declaration marked
+`base` or `final` is not also marked `base`, `final`, or `sealed`.
+This restriction applies to both direct and indirect subtypes
+and along all paths that introduce subtypes:
 `implements` clauses, `extends` clauses, `with` clauses, and `on` clauses. This
-restriction applies even to types within the same library.
+restriction applies even to types within the same library._
```

Comment:
After several readings, I get what three previous paragraphs are saying, but I find the name and the use of "interface" pretty confusing. An "interface" that you're prevented implement is... what how is that an "interface" at all?

The fact that the third paragraph is needed to translate it to something comprehensible is probably telling. :)

How about:

> We say that a type is *unimplementable* if any of its supertypes are marked `base` or `final`.
>
> It is a compile-time error if a `class`, `mixin`, or `mixin class` declaration is unimplementable and doesn't itself have a `base`, `final` or `sealed` modifier.  _(Which must be a `base` modifier for `mixin` and `mixin class` declarations.)_
>
> _This restriction applies along all paths that introduce subtypes: `implements` clauses, `extends` clauses, `with` clauses, and `on` clauses. This restriction applies even to types within the same library._

?

I understand that "interface" is more spec-like terminology for "static type-y thing that has declarations", but it's a really confusing term in this context and something less formal but still accurate might be clearer, especially in a proposal that is literally adding an `interface` modifier.